auth/api/iam/api.go

```diff
@@ -228,6 +230,43 @@ func (r Wrapper) GetOAuthAuthorizationServerMetadata(ctx context.Context, reques
 	return GetOAuthAuthorizationServerMetadata200JSONResponse(authorizationServerMetadata(*identity)), nil
 }
 
+// GetOAuthClientMetadata returns the OAuth2 Client metadata for the request.Id if it is managed by this node.
+func (r Wrapper) GetOAuthClientMetadata(ctx context.Context, request GetOAuthClientMetadataRequestObject) (GetOAuthClientMetadataResponseObject, error) {
+	if err := r.validateAsNutsFingerprint(ctx, request.Id); err != nil {
+		return nil, fmt.Errorf("client metadata: %w", err)
+	}
+
+	identity := r.auth.PublicURL().JoinPath("iam", request.Id)
+
+	return GetOAuthClientMetadata200JSONResponse(clientMetadata(*identity)), nil
+}
+
+func (r Wrapper) validateAsNutsFingerprint(ctx context.Context, fingerprint string) error {
+	// convert fingerprint to did:nuts
+	if strings.HasPrefix(fingerprint, "did:") {
+		return core.InvalidInputError("id contains full did")
+	}
+	nutsDID, err := did.ParseDID("did:nuts:" + fingerprint)
+	if err != nil {
+		return core.InvalidInputError(err.Error())
+	}
+
+	// assert ownership of did
+	owned, err := r.vdr.IsOwner(ctx, *nutsDID)
```

Comment:
I'm guessing IsOwner doesn't work on web dids since we have did:nuts in filename/vaultKey

Otherwise converting to nuts did is weird.

We should probably fix IsOwner to work with web dids

Reply:
correct. currently a did:web is just a derivative of did:nuts.